cost-analyzer/templates/cost-analyzer-frontend-config-map-template.yaml

```diff
@@ -884,17 +885,24 @@ data:
             default_type 'application/json';
             add_header 'Access-Control-Allow-Origin' '*' always;
             add_header 'Access-Control-Allow-Methods' 'GET, PUT, POST, DELETE, OPTIONS' always;
-            {{- if and .Values.diagnostics.enabled .Values.diagnostics.isDiagnosticsPrimary.enabled }}
+            {{- if and .Values.diagnostics.enabled .Values.diagnostics.primary.enabled }}
             {{- if (not (empty .Values.kubecostModel.federatedStorageConfigSecret )) }}
             return 200 '{"multi-cluster-diagnostics-enabled": "true"}';
             {{- end }}
             {{- else }}
             return 200 '{"multi-cluster-diagnostics-enabled": "false"}';
             {{- end }}
         }
-        {{- if and .Values.diagnostics.enabled .Values.diagnostics.isDiagnosticsPrimary.enabled }}
+
+        # When the Multi-cluster Diagnostics Service is run within the
+        # cost-model container, its endpoint is available at
+        # `/model/diagnostics/multicluster`. No Nginx path forwarding needed.
+        # When the Multi-cluster Diagnostics Service is run as a K8s Deployment,
+        # we should forward that path to the K8s Service.
+
+        {{- if and .Values.diagnostics.enabled .Values.diagnostics.primary.enabled .Values.diagnostics.deployment.enabled }}
         {{- if (not (empty .Values.kubecostModel.federatedStorageConfigSecret )) }}
-        location /model/multi-cluster-diagnostics {
+        location /model/diagnostics/multicluster {
```

Comment:
Could we possibly put this service under its own upstream? `/diagnostics` or something?

Reply:
that's taken, what do you think of /diagnostics/api

Reply:
I believe Thomas is OOO today, FYI.

Reply:
ya, I'll get this merged

Reply:
I can see the reasoning for why we would make this path `/diagnostics/api` since the API is being served by the Diagnostics k8s service and not Cost-model.

However, I still favor making the Nginx location `/model/diagnostics/multicluster`. This allows the frontend/user to always make API requests to `/model/diagnostics/multicluster` regardless of how Multi-cluster diagnostics has been deployed. Otherwise the user would need to make API requests to `/model/diagnostics/multicluster` if Diagnostics is running within cost-model ([code ref](https://github.com/kubecost/kubecost-cost-model/pull/2163/files#diff-c9127ce2d87ae5fe988d6b5f16b63f7ba022998d617dfbafd0381bfda6ceba41R1712)), and then requests to `/diagnostics/api` if Diagnostics is running as a separate Deployment.